src/orion/algo/hyperband.py

```diff
@@ -497,56 +570,65 @@ def seed(self, seed):
         self._seed = seed
         self._samples = None
 
-    def get_sample(self):
+    def get_sample(self) -> Trial | None:
         if self._samples is None:
             n_samples = int(self.rungs[0]["n_trials"] * self.buffer)
             self._samples = self.hyperband.space.sample(n_samples, seed=self.seed)
 
         return self._samples.pop(0) if self._samples else None
 
-    def sample(self, num):
+    def sample(self, num: int) -> list[Trial]:
         """Sample a new trial with lowest fidelity"""
         should_have_n_trials = self.rungs[0]["n_trials"]
         n_trials = len(self.rungs[0]["results"])
         request = max(min(should_have_n_trials - n_trials, num), 0)
         if request == 0:
             return []
-
+        # BUG: Shouldn't this be `sample_from_bracket`?
```

Comment:
It might be that `request` is always equal to 0, since we never call `suggest` with a big enough number. 
In any case, I'd argue that it is out of the scope of this PR to do this kind of debugging + refactoring.

Reply:
Agreed! Created #857